apps/restconf/restconf_methods.c

```diff
@@ -581,6 +581,367 @@ api_data_write(clicon_handle h,
+    memset(s, 0, TEMP_STR_MALLOC_SIZE);
+    return s;
+}
+
+int cpy_str(char *dest, char *src, size_t size)
+{
+    if (src == NULL) {
+        return 0;
+    }
+    if (dest == NULL) {
+        init_str(dest);
+    }
+    if (size <= 0 || size > TEMP_STR_MALLOC_SIZE) {
+        return 0;
+    }
+    size_t i;
+    for (i = 0; i < size - 1 && src[i]; i++) {
+        dest[i] = src[i];
+    }
+    dest[i] = '\0';
+    return i;
+}
+
+int cat_str(char *dest, char *src, size_t size)
+{
+    if (src == NULL) {
+        return 0;
+    }
+    if (dest == NULL) {
+        init_str(dest);
+    }
+    if (size <= 0 || size > TEMP_STR_MALLOC_SIZE) {
+        return 0;
+    }
+    size_t i;
+    int old_len = strlen(dest);
+    for (i = 0; i < size - 1 && src[i]; i++) {
+        dest[i + old_len] = src[i];
+    }
+    dest[i + old_len] = '\0';
+    return i;
+}
+
+void free_mem(void *str)
+{
+    if (str != NULL)
+        free(str);
+}
+
+int get_xval(
+    cvec* nsc,
+    cxobj* xn,
+    char* val,
+    cxobj **vec,
+    size_t veclen,
+    const char* key
+    )
+{
+    char* tmp_val = NULL;
+    int ret = xpath_vec(xn, nsc, "%s", &vec, &veclen, key);
+    if (ret < 0) {
+        return ret;
+    }
+    for (int j = 0; j < veclen; j++) {
+        cxobj *xn = vec[j];
+        tmp_val = xml_body(xn);
+    }
+    cpy_str(val, tmp_val, TEMP_STR_MALLOC_SIZE);
+    return 0;
+}
+
+int do_replace (
+        clicon_handle h,
+	    void         *req,
+	    int           pi,
+	    cvec         *qvec,
+	    int           pretty,
+	    restconf_media media_out,
+	    ietf_ds_t     ds,
+        char* simple_patch_request_uri,
+        char* target_val,       
+        int value_vec_len,
+        cxobj** value_vec,
+        cxobj * value_vec_tmp,
+        cxobj *x_simple_patch,
+        char *patch_header
+      )
+{
+      char *delete_req_uri = init_str();
+      if (delete_req_uri == NULL)
+          return 1;
+
+      if (cpy_str(delete_req_uri, simple_patch_request_uri, TEMP_STR_MALLOC_SIZE) <= 0)
+          return 1;
+
+      if (cat_str(delete_req_uri, target_val, TEMP_STR_MALLOC_SIZE) <= 0)
+          return 1;
+
+      // Delete the object with the old values
+      int ret = api_data_delete(h, req, delete_req_uri, pi, pretty, YANG_DATA_JSON, ds );
+      free_mem((void *)delete_req_uri);
+      if (ret != 0)
+          return ret;
+
+      // Now insert the object with the new values
+      char *json_simple_patch = init_str();
+      if (json_simple_patch == NULL)
+          return 1; // goto done;
+
+      for (int k = 0; k < value_vec_len; k++) {
+          if (value_vec[k] != NULL) {
+              value_vec_tmp = xml_dup(value_vec[k]);
+              xml_addsub(x_simple_patch, value_vec_tmp);
+          }
+      }
+      cbuf* cb = cbuf_new();
+      xml2json_cbuf(cb, x_simple_patch, 1);
+
+      // Some ugly text processing to get the JSON to match what api_data_post() expects
+      char *json_simple_patch_tmp = cbuf_get(cb);
```

Comment:
Can we modify api_data_post to support this syntax? 
In any case, I would suggest adding more explicit description to hacks.